File: Java_ChildrenCompetition/src/main/java/repository/UserDbRepository.java
package repository;

import model.User;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.Properties;

public class UserDbRepository implements IUserRepository<Integer, User>{
    private User currentUser;
    private final JdbcUtils dbUtils;
    private static final Logger logger = LogManager.getLogger();

    public UserDbRepository(Properties properties) {
        logger.info("initializing UserDbRepository with properties: {} ", properties);
        dbUtils = new JdbcUtils(properties);
        currentUser = null;
    }

    @Override
    public void setCurrentUser(User entity) {
        this.currentUser = entity;
    }

    @Override
    public User getCurrentUser() {
        return currentUser;
    }


    @Override
    public int size() {
        return 0;
    }

    @Override
    public void save(User user) {
        logger.traceEntry("saving task {} ", user);
        Connection connection = dbUtils.getConnection();
        String query = "INSERT into users(firstname, lastname, username, password) values (?,?,?,?)";

        try(PreparedStatement preparedStatement = connection.prepareStatement(query)){
            preparedStatement.setString(1, user.getFirstname());
            preparedStatement.setString(2, user.getLastname());
            preparedStatement.setString(3, user.getUsername());
            preparedStatement.setString(4, user.getPassword());

            int result = preparedStatement.executeUpdate();
            logger.trace("saved {} instances", result);

        } catch (SQLException exception){
            logger.error(exception);
            System.err.println("Error DB " + exception);
        }
        logger.traceExit();
    }

    @Override
    public void delete(Integer id) {
        logger.traceEntry("deleting task with {} ", id);
        Connection connection = dbUtils.getConnection();
        String query = "DELETE from users where id = ?";

        try(PreparedStatement preparedStatement = connection.prepareStatement(query)) {
            preparedStatement.setInt(1, id);
            int result = preparedStatement.executeUpdate();

        } catch (SQLException exception){
            logger.error(exception);
            System.err.println("Error DB " + exception);
        }
        logger.traceExit();

    }

    @Override
    public void update(Integer integer, User entity) {

    }

    @Override
    public User findOne(Integer id) {
        logger.traceEntry("finding task with id {} ", id);
        Connection connection = dbUtils.getConnection();
        String query = "SELECT * from users where id_user = ?";

        try(PreparedStatement preparedStatement = connection.prepareStatement(query)){
            preparedStatement.setInt(1, id);
            try(ResultSet resultSet = preparedStatement.executeQuery()){
                if(resultSet.next()) {
                    int idUser = resultSet.getInt("id_user");
                    String firstname = resultSet.getString("firstname");
                    String lastname = resultSet.getString("lastname");
                    String username = resultSet.getString("username");
                    String password = resultSet.getString("password");
                    User user = new User(firstname, lastname, username, password);
                    user.setId(idUser);
                    logger.traceExit(user);
                    return user;
                }
            }
        } catch (SQLException exception){
            logger.error(exception);
            System.err.println("Error DB " + exception);
        }

        return null;
    }

    @Override
    public User findByUsername(String username) {
        logger.traceEntry("finding task with username {} ", username);
        Connection connection = dbUtils.getConnection();
        String query = "SELECT * from users where username = ?";

        try(PreparedStatement preparedStatement = connection.prepareStatement(query)){
            preparedStatement.setString(1, username);
            try(ResultSet resultSet = preparedStatement.executeQuery()){
                if(resultSet.next()) {
                    int idUser = resultSet.getInt("id_user");
                    String firstname = resultSet.getString("firstname");
                    String lastname = resultSet.getString("lastname");
                    String usernameUser = resultSet.getString("username");
                    String password = resultSet.getString("password");
                    User user = new User(firstname, lastname, usernameUser, password);
                    user.setId(idUser);
                    logger.traceExit(user);
                    return user;
                }
            }
        } catch (SQLException exception){
            logger.error(exception);
            System.err.println("Error DB " + exception);
        }

        return null;
    }

    @Override
    public Iterable<User> findAll() {
        return null;
    }
}

File: CSharp_ChildrenCompetitionSockets/CSharp_ChildrenCompetitionSockets/persistence/UserDbRepository.cs
using System;
using System.Collections.Generic;
using System.Data;
using CSharp_ChildrenCompetitionGUI.model;
// using log4net;

namespace CSharp_ChildrenCompetitionGUI.repository
{
    public class UserDbRepository : IUserRepository
    {
        private User currentUser;
        
        // private static readonly ILog log = LogManager.GetLogger("UserDbRepository");

        // private IDictionary<String, string> props;

        public UserDbRepository(IDictionary<String, string> props)
        {
            // log.Info("Creating UserDbRepository");
            // this.props = props;
        }
        
        public void setCurrentUser(User entity)
        {
            // throw new System.NotImplementedException();
            this.currentUser = entity;
        }

        public User getCurrentUser()
        {
            // throw new System.NotImplementedException();
            return this.currentUser;
        }
        
        public int size()
        {
            throw new System.NotImplementedException();
        }

        public void save(User entity)
        {
            // throw new System.NotImplementedException();
            var conn = DBUtils.getConnection();

            using (var command = conn.CreateCommand())
            {
                command.CommandText = "INSERT into users(firstname, lastname, username, password) values (@fn,@ln,@un,@pw)";
                var paramFN = command.CreateParameter();
                paramFN.ParameterName = "@fn";
                paramFN.Value = entity.firstname;
                command.Parameters.Add(paramFN);

                var paramLN = command.CreateParameter();
                paramLN.ParameterName = "@ln";
                paramLN.Value = entity.lastname;
                command.Parameters.Add(paramLN);

                var paramUN = command.CreateParameter();
                paramUN.ParameterName = "@un";
                paramUN.Value = entity.username;
                command.Parameters.Add(paramUN);

                var paramPW = command.CreateParameter();
                paramPW.ParameterName = "@pw";
                paramPW.Value = entity.password;
                command.Parameters.Add(paramPW);

                var result = command.ExecuteNonQuery();
            }
        }

        public void delete(int id)
        {
            throw new System.NotImplementedException();
        }

        public void update(int id, User entity)
        {
            throw new System.NotImplementedException();
        }

        public User findOne(int id)
        {
            // throw new System.NotImplementedException();
            // log.InfoFormat("Find one with value {0}", id);
            IDbConnection conn = DBUtils.getConnection();

            using (var comm = conn.CreateCommand())
            {
                comm.CommandText = "SELECT * from users where id_user = @id";
                IDbDataParameter paramId = comm.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = id;
                comm.Parameters.Add(paramId);

                using (var dataR = comm.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        int idU = dataR.GetInt32(0);
                        String firstname = dataR.GetString(1);
                        String lastname = dataR.GetString(2);
                        String username = dataR.GetString(3);
                        String password = dataR.GetString(4);
                        User user = new User(firstname, lastname, username, password);
                        user.id = idU;
                        // log.InfoFormat("Exiting findOne with value{0}", user);
                        return user;
                    }
                }
            }
            // log.InfoFormat("exiting findOne with value{0}", null);
            return null;

        }

        public IEnumerable<User> findAll()
        {
            throw new System.NotImplementedException();
        }

        public User findByUsername(string username)
        {
            // throw new System.NotImplementedException();
            // throw new System.NotImplementedException();
            // log.InfoFormat("Find one with value {0}", username);
            IDbConnection conn = DBUtils.getConnection();

            using (var comm = conn.CreateCommand())
            {
                comm.CommandText = "SELECT * from users where username = @us";
                IDbDataParameter paramId = comm.CreateParameter();
                paramId.ParameterName = "@us";
                paramId.Value = username;
                comm.Parameters.Add(paramId);

                using (var dataR = comm.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        int id = dataR.GetInt32(0);
                        String firstname = dataR.GetString(1);
                        String lastname = dataR.GetString(2);
                        String usernameU = dataR.GetString(3);
                        String password = dataR.GetString(4);
                        User user = new User(firstname, lastname, usernameU, password);
                        user.id = id;
                        // log.InfoFormat("Exiting findOne with value{0}", user);
                        return user;
                    }
                }
            }
            // log.InfoFormat("exiting findOne with value{0}", null);
            return null;
        }
    }
}